types/serialization.go

```diff
@@ -80,17 +89,24 @@ func (h *SignedHeader) FromProto(other *pb.SignedHeader) error {
 		return err
 	}
 
-	err = h.Validators.FromProto(other.Validators)
-	if err != nil {
-		return err
-	}
+	if other.Validators != nil && other.Validators.GetProposer() != nil {
```

Comment:
you don't need to check for `nil`. it is handled automatically in `ValidatorSetFromProto`
```
h.Validators, err = types.ValidatorSetFromProto(other.Validators)
if err != nil {
     return err
}
```

Reply:
Test